draft-wood-tls-external-psk-importer.md

```diff
@@ -97,20 +100,29 @@ supported HashAlgorithm 'hash', the importer constructs an ImportedIdentity stru
    } ImportedIdentity;
 ~~~
 
+ImportedIdentity.label SHOULD be bound to the protocol for which the key is imported. For example,
+TLS 1.3 over TCP should use "tls13" as the label, whereas QUIC {{!I-D.ietf-quic-transport}} should use
+"quicv1" as the label. Note that this means future versions of TLS will increase the number of
+PSKs derived from an external PSK.
+
 A unique and imported PSK (IPSK) with base key 'ipskx' bound to this identity is then computed as follows:
 
 ~~~
    epskx = HKDF-Extract(0, epsk)
-   ipskx = HKDF-Expand-Label(epskx, "derived psk", Hash(ImportedIdentity), Hash.length)
+   ipskx = HKDF-Expand-Label(epskx, "derived psk",
+                             Hash(ImportedIdentity), Hash.length)
 ~~~
 
 The hash function used for HKDF {{!RFC5869}} is that which is associated with the external PSK. It is not
 bound to ImportedIdentity.hash. If no hash function is specified, SHA-256 MUST be used.
+Differentiating epsk by ImportedIdentity.hash ensures that each imported PSK is only used with at most one
+hash function, thus satisfying the requirements in {{!RFC8446}}. Endpoints MUST import and derive an
+ipsk for each ciphersuite they support. For example, importing a key for TLS_AES_128_GCM_SHA256 and
```

Comment:
Nit: maybe instead "Endpoints MUST import and derive an ipsk for each hash function used by each ciphersuite they support.". If I offer TLS_AES_128_GCM_SHA256 and TLS_CHACHA20_POLY1305_SHA256, I don't need two of them.

Reply:
Fixed!